src/main/java/teammates/ui/webapi/action/Action.java

```diff
@@ -225,11 +226,11 @@ protected String getRequestBody() {
     /**
      * Checks the specific access control needs for the resource.
      */
-    public abstract void checkSpecificAccessControl();
+    public abstract void checkSpecificAccessControl() throws InvalidParametersException;
 
     /**
      * Executes the action.
      */
-    public abstract ActionResult execute();
+    public abstract ActionResult execute() throws InvalidParametersException;
```

Comment:
These two methods should not throw anything, reason: `InvalidParametersException` (and other `Exception`s) can come from many different sources and the treatment for each can be different.

Specifically for this issue, if the decryption fail, it means the input from the front-end is bad, so it is appropriate to re-throw as `InvalidHttpParameterException`,